DependencyInjection/Configuration.php

```diff
@@ -37,6 +45,9 @@ public function getConfigTreeBuilder()
                 ->booleanNode('use_table_diffing')
                     ->defaultTrue()
                 ->end()
+                ->scalerNode('purifier_cache_location')
+                    ->defaultValue($this->kernel->getCacheDir())
```

Comment:
I think it would be better to avoid injecting the kernel into the Configuration class - instead, we should just inject the cache directory string, and we can do that through the `CaxyHtmlDiffExtension` class.

So in this class, we should have a `protected $cacheDir;`  property, and the constructor will take $cacheDir.

Then you'll need to update the `CaxyHtmlDiffExtension` function to get the cache dir from the container and inject it where the `Configuration` instance is being created in the `load` function:

```php
// in HtmlDiffBundle/DependencyInjection/CaxyHtmlDiffExtension.php

// ...

public function load(array $configs, ContainerBuilder $container)
{
    $configuration = new Configuration($container->getParameter('kernel.cache_dir');

    // ...
}
```

We will then need to add a method call to the caxy.html_diff.config service in the Extension to pass in the config variable:

```php
// in HtmlDiffBundle/DependencyInjection/CaxyHtmlDiffExtension.php

// ...

protected function loadHtmlDiffConfig(array $config, ContainerBuilder $container, $cacheDriverId = null)
{
    // ...

    $methodsToCall = array(
        // ...
        'purifier_cache_location' => 'setPurifierCacheLocation`,
    );
}
```

That should be all that's needed! It's important to inject only the dependencies that are required.

**Note**: You could also do all of this in the Extension class - you could simply set the purifier_cache_location setting to `->defaultNull()` and then in this Extension class, in the `loadHtmlDiffConfig` function you could do

```php
if (!isset($config['purifier_cache_location']) {
    $config['purifier_cache_location'] = $container->getParameter('kernel.cache_dir');
}
```

There are pros and cons to the two methods... I actually prefer this second method, since it only accesses the kernel.cache_dir parameter when it's needed, as opposed to loading it every time in order to instantiate the `Configuration` object. It's also one less dependency that way.

The only advantage to setting the default in the `Configuration` class is that you can see the default value when dumping the config reference through `app/console`. So I still the other method.